starter_code/src/movies.js

```diff
@@ -1,20 +1,117 @@
+// ==========================================================
 /* eslint no-restricted-globals: 'off' */
-// Turn duration of the movies from hours to minutes 
+// Turn duration of the movies from hours to minutes
+function turnHoursToMinutes(arr){
+	let hour = arr.duration.replace(/h/g, "").replace(/min/g, "").split(" ").map(function(el){ return parseInt(el) })[0]
+	let min = arr.duration.replace(/h/g, "").replace(/min/g, "").split(" ").map(function(el){ return parseInt(el) })[1]
+	let result = (hour * 60 + min).toString()
+	arr.duration = result
+	return arr
+}
 
 
+let newMovies = movies.map(function(el){
+	return turnHoursToMinutes(el)
+})
+
+console.log(newMovies)
+
+
+
+// ==========================================================
+
 // Get the average of all rates with 2 decimals 
+function ratesAverage(moviesArr){
+  let result = moviesArr.reduce(function(accumulator, el){
+    return parseFloat(el.rate) + accumulator
+  }, 0)
+  return parseFloat((result / moviesArr.length).toFixed(2));
+}
+// test
+ratesAverage(movies)
 
 
-// Get the average of Drama Movies
 
+// ==========================================================
 
+// Get the average of Drama Movies
+function dramaMoviesRate(arrayRate){
+  let filteredArray = arrayRate.filter(function(el){
+    return (el.genre.indexOf("Drama") !== -1)
+  });
```

Comment:
this works perfect, but there is also a new method called "includes" that makes unnecessary to use this little trick, you can do the same thing by saying : 
```
return (el.genre.includes("Drama"))
```